base/protocol.py

```diff
@@ -31,17 +32,19 @@ class Prove(bt.Synapse):
         description="The polynomial to prove.",
         allow_mutation=False,
     )
-    commitment: Optional[str] = Field(
-        title="Commitment", description="The commitment to the polynomial.", default=None,
+    x: str = Field(
+        title="Input",
+        description="The input to evaluate the polynomial at.",
+        default=None,
```

Comment:
I believe this might crash if we instantiate a synapse where we dont specify `x` since `None` isn't a `str`